### PR TITLE
Move theming support from Beta to Main

### LINKED ISSUE #95: Update color tokens for theming support
### General Support Request

In order to make Auro components themeable, we need to make sure there are no hard-coded and correct semanticly named tokens within the component. There also needs to be investigation around `border` rules which have a combination of size and color tokens, making them more complex, as well as `background-color` and `color` rules.

### Possible Solution

Work with designers to figure out what tokens need updated values or renamed.

### Additional context

This work stems from theming work happening to all Auro components:
- https://github.com/AlaskaAirlines/AuroDesignTokens/issues/147

### Acceptance Criteria:

- [ ] All tier 3 token names have been implemented.
- [ ] Colors identified below have been updated to their tier 3 token name.

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #95 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Move theming support for the auro-badge component from beta to main, enabling customization through design tokens. Refactor the component to utilize a new dependency versioning system and enhance its functionality with new methods for handling SVG icons. Update documentation and tests to align with these changes, and clean up deprecated content.

New Features:
- Introduce theming support for the auro-badge component, allowing customization of styles using design tokens.

Enhancements:
- Refactor the auro-badge component to use a new dependency versioning system for auro-button and auro-icon elements.
- Improve the auro-badge component by adding a method to generate HTML elements with SVG icons.

Documentation:
- Update documentation to include new theming support details and modify API documentation paths.

Tests:
- Update tests to reflect changes in the auro-badge component, specifically targeting the new button element.

Chores:
- Remove deprecated CSS custom property fallbacks from the README and related documentation.

### CODE REVIEW

File: src/tokens.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import "./../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables";
+
+:host {
+  --ds-auro-badge-border-color: var(--ds-color-border-subtle-inverse, #{$ds-color-border-subtle-inverse});
+  --ds-auro-badge-boxshadow-color: var(--ds-color-border-subtle-default, #{$ds-color-border-subtle-default});
```
Comment (jordanjones243) [inline src/tokens.scss:5]: Boxshaodw using default and border using inverse? Investigate more

File: README.md [hunk 2]
```diff
@@ -82,15 +82,6 @@ The use of any Auro custom element has a dependency on the [Auro Design Tokens](
...
 <!-- AURO-GENERATED-CONTENT:END -->
 
-### CSS Custom Property fallbacks
-
-<!-- AURO-GENERATED-CONTENT:START (REMOTE:url=https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/partials/usage/cssFallbacks.md) -->
-[CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) are [not supported](https://auro.alaskaair.com/support/custom-properties) in older browsers. For this, fallback properties are pre-generated and included with the npm.
-
-Any update to the Auro Design Tokens will be immediately reflected with browsers that support CSS custom properties, legacy browsers will require updated components with pre-generated fallback properties.
-
-<!-- AURO-GENERATED-CONTENT:END -->
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply README.md:92]: This has been done in every component.

File: src/auro-badge.js [hunk 3]
```diff
@@ -94,19 +111,59 @@ export class AuroBadge extends LitElement {
...
+   */
+  generateIconHtml(svgContent) {
+    const dom = new DOMParser().parseFromString(svgContent, 'text/html');
+    const svg = dom.body.firstChild;
+
+    svg.setAttribute('slot', 'svg');
+
+    const iconHtml = html`<${this.iconTag} customColor customSize customSvg slot="icon">${svg}</${this.iconTag}>`;
+
+    return iconHtml;
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply src/auro-badge.js:150]: Good idea to simplify, will address later.

File: src/tokens.scss (hunk 1 above)
Comment (jordanjones243) [reply src/tokens.scss:5]: Boxshadow is using what I suspect to be the correct color.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**